### PR TITLE
CMake CS FMUs with CVODE

### PR DESCRIPTION
**Pull request by AnHeuermann**
### Related Issues

CMake FMUs for CoSimulation couldn't be build when choosing CVODE as integration method.

### Purpose

Fix it.

### Approach

  - `CVODE_DIRECTORY` now uses `/lib/${CMAKE_LIBRARY_ARCHITECTURE}/omc`
  - Made tests have shorter names for Windows
  - Adding CVODE + CMake test for CoSimulation FMUs
  - Updating failing test simpleStiffFMU

### CODE REVIEW

File: OMCompiler/SimulationRuntime/fmi/export/buildproject/CMakeLists.txt.in [hunk 1]
```diff
@@ -76,6 +76,9 @@ set(CMAKE_SHARED_LIBRARY_PREFIX "")
 if(APPLE)
   set(CMAKE_INSTALL_RPATH "@loader_path")
 else()
+  if(${NEED_CVODE} AND RUNTIME_DEPENDENCIES_LEVEL STREQUAL "none")
+    set(CMAKE_INSTALL_RPATH_USE_LINK_PATH True)
+  endif()
```
[the remark this answers is not in the document]
Comment (mahge) [reply OMCompiler/SimulationRuntime/fmi/export/buildproject/CMakeLists.txt.in:81]: I am not sure if I understood everything but the first question is why are we linking `sundials_cvode` dynamically? We do not even build the dynamic version with the CMake config. The autoconf build builds both so we can use the static one of needed. 

If we need to link the dynamic version you have to tell me more :) I am not sure what the issue is. Is there something special for `sundials_cvode` compared to say `libc.so`?

[the remark this answers is not in the document]
Comment (AnHeuermann) [reply OMCompiler/SimulationRuntime/fmi/export/buildproject/CMakeLists.txt.in:81]: In `buildModelFMU` we can specify `platforms={"dynamic"}`. Currently the CMake FMU build doesn't really use that information, but I think in that case it should use the shared version of `sundials_cvode`. BUt you are right, we don't compile that one for the CMake build of OpenModelica.
Maybe I should try to prefer the static version of CVODE and just link that one if possible.

[the remark this answers is not in the document]
Comment (AnHeuermann) [reply OMCompiler/SimulationRuntime/fmi/export/buildproject/CMakeLists.txt.in:81]: The issue is, that if we want to use the dynamic version of `sundials_cvode` we need to copy it into the FMU, but for that we need a newer CMake than our CI provides. Basically this is is a workaround, to link the dynamic version of without copying into the FMU.

[the remark this answers is not in the document]
Comment (AnHeuermann) [reply OMCompiler/SimulationRuntime/fmi/export/buildproject/CMakeLists.txt.in:81]: Okay. I think this PR should be fine for now.
If someone complains we can still change it.

**Pull request merged**